src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -1265,28 +1267,20 @@ public Page search(String q) throws Exception {
     }
 
     /**
-     * Get JSON from A Jenkins endpoint.
-     * @param path The endpoint URL.
-     * @return The JSON.
+     * Get JSON from a Jenkins relative endpoint. Create a new default webclient. If you want to configure the
+     * webclient, for example to set a token for authentication, or accept other HTTP responses than 200, you can use
+     * {@link WebClient#getJSON(String)} directly.
+     *
+     * @param path relative path, should not start with '/'
+     * @return The JSON response from server.
      */
     public JSONWebResponse getJSON(@NonNull String path) throws IOException {
-        assert !path.startsWith("/");
-
         JenkinsRule.WebClient webClient = createWebClient();
-        Page runsPage = null;
-        try {
-            runsPage = webClient.goTo(path, "application/json");
-        } catch (SAXException e) {
-            // goTo shouldn't be throwing a SAXException for JSON.
-            throw new IllegalStateException("Unexpected SAXException.", e);
-        }
-        WebResponse webResponse = runsPage.getWebResponse();
-
-        return new JSONWebResponse(webResponse);
+        return webClient.getJSON(path);
     }
 
     /**
-     * POST a JSON payload to a URL on the underlying Jenkins instance.
+     * POST a JSON payload to a URL on the underlying Jenkins instance using the crumb.
```

Comment:
Probably want to link to `WebClient.postJSON` as a more straightforward option, noting that it does not automatically pass a crumb (if there is no `CrumbExclusion` you can `createCrumbedURL` or just pass an API token). I am not sure what the other complexity in this old method is about; seems unnecessary.

Reply:
Yes the old method seems a bit complex, but it would requires investigations on usages to change it IMO. I don't want to touch it :D